BeagleBone_black/src/motor_control/controller.cpp

```diff
@@ -41,55 +48,66 @@ using utils::concurrent::Thread;
  * Bytes 4-7: Data
  */
 
-/* NMT CAN Frames composed as follows:
+/* The Network Management (NMT) Protocol is used to start/stop and reset the Controller nodes in the system.
+ * NMT CAN Frames composed as follows:
  * CAN ID: 0x000
  * Len:    2
  * Byte 1: NMT Function
  * Byte 2: Target Node
  */
 
 // Types of CANopen messages, these are used for CAN ID's
-constexpr uint32_t kEMGY_TRANSMIT          = 0x80;
-constexpr uint32_t kSDO_RECEIVE            = 0x600;
-constexpr uint32_t kSDO_TRANSMIT           = 0x580;
-constexpr uint32_t kNMT_RECEIVE            = 0x000;
-constexpr uint32_t kNMT_TRANSMIT           = 0x700;
+constexpr uint32_t kEmgyTransmit          = 0x80;
+constexpr uint32_t kSdoReceive            = 0x600;
+constexpr uint32_t kSdoTransmit           = 0x580;
+constexpr uint32_t kNmtReceive            = 0x000;
+constexpr uint32_t kNmtTransmit           = 0x700;
 
 // Function codes for sending messages
-constexpr uint8_t  kREAD_OBJECT            = 0x40;
-constexpr uint8_t  kWRITE_1_BYTE           = 0x2F;
-constexpr uint8_t  kWRITE_2_BYTES          = 0x2B;
-constexpr uint8_t  kWRITE_3_BYTES          = 0x27;
-constexpr uint8_t  kWRITE_4_BYTES          = 0x23;
+constexpr uint8_t  kReadObject            = 0x40;
+constexpr uint8_t  kWriteOneByte          = 0x2F;
+constexpr uint8_t  kWriteTwoBytes         = 0x2B;
+constexpr uint8_t  kWriteThreeBytes       = 0x27;
+constexpr uint8_t  kWriteFourBytes        = 0x23;
 
 // Network management commands
-constexpr uint8_t  kNMT_OPERATIONAL        = 0x01;
-constexpr uint8_t  kNMT_STOP               = 0x02;
-constexpr uint8_t  kNMT_PREOPERATIONAL     = 0x80;
-constexpr uint8_t  kNMT_RESET_NODE         = 0x81;
-constexpr uint8_t  kNMT_RESET_COMMS        = 0x82;
+constexpr uint8_t  kNmtOperational        = 0x01;
+constexpr uint8_t  kNmtStop               = 0x02;
+constexpr uint8_t  kNmtPreOperational     = 0x80;
+constexpr uint8_t  kNmtResetNode          = 0x81;
+constexpr uint8_t  kNmtResetComms         = 0x82;
 
 Controller::Controller(Logger& log, uint8_t id)
   : log_(log),
     can_(Can::getInstance()),
     data_(data::Data::getInstance()),
+    motor_data_(data_.getMotorData()),
```

Comment:
Are you aware that `data_.getMotorData()` contains random garbage (except for status which is `kStart`) at this point?

Reply:
Thanks, should be fixed now